src/fetch_package.js

```diff
@@ -82,7 +132,20 @@ export default async function fetchPackage(
             }
 
             if (requestStats) {
-                requestStats.packageFetches++;
+                /**
+                 * A little state machine to track what came from the cache.
+                 */
+                if (result._cached === "new") {
+                    result._cached = "no";
+                } else if (result._cached === "no") {
+                    result._cached = "yes";
+                }
+
+                if (result._cached === "yes") {
+                    requestStats.fromCache++;
+                } else {
```

Comment:
This bit won't work due to the cloning that the cache performs. However, the basics are sound. Since we can manipulate what is in the cache before its cached, I think we can just check for a token (timestamp match, or whatever) that would be unique per fetch call, and therefore, if it doesn't match, it was from the cache.

Reply:
(also, yay for multiline comments)